File: frontend/src/services/marketplaceService.ts
import apiClient from './apiClient';
import {
    TemplateDto,
    TemplateDetailDto,
    CreateTemplateRequest,
    UpdateTemplateRequest,
    InstallTemplateRequest,
    InstallationResultDto,
    RateTemplateRequest,
    AddCommentRequest,
    CommentDto,
    TemplateSearchRequest,
    TemplateStatsDto,
    CategoryDto,
    TagDto,
    OrganizationDto,
    OrganizationDetailDto,
    Page,
} from '../types/marketplace';

class MarketplaceService {
    private basePath = '/marketplace';

    // Template endpoints
    async searchTemplates(params: TemplateSearchRequest): Promise<Page<TemplateDto>> {
        const response = await apiClient.get(`${this.basePath}/templates`, { params });
        return response.data;
    }

    async getTemplateDetails(slug: string): Promise<TemplateDetailDto> {
        const response = await apiClient.get(`${this.basePath}/templates/${slug}`);
        return response.data;
    }

    async createTemplate(request: CreateTemplateRequest): Promise<TemplateDto> {
        const response = await apiClient.post(`${this.basePath}/templates`, request);
        return response.data;
    }

    async updateTemplate(slug: string, request: UpdateTemplateRequest): Promise<TemplateDto> {
        const response = await apiClient.put(`${this.basePath}/templates/${slug}`, request);
        return response.data;
    }

    async uploadIcon(slug: string, file: File): Promise<void> {
        const formData = new FormData();
        formData.append('file', file);
        await apiClient.post(`${this.basePath}/templates/${slug}/icon`, formData, {
            headers: { 'Content-Type': 'multipart/form-data' },
        });
    }

    async addScreenshot(slug: string, file: File): Promise<void> {
        const formData = new FormData();
        formData.append('file', file);
        await apiClient.post(`${this.basePath}/templates/${slug}/screenshots`, formData, {
            headers: { 'Content-Type': 'multipart/form-data' },
        });
    }

    async installTemplate(slug: string, request: InstallTemplateRequest): Promise<InstallationResultDto> {
        const response = await apiClient.post(`${this.basePath}/templates/${slug}/install`, request);
        return response.data;
    }

    async rateTemplate(slug: string, request: RateTemplateRequest): Promise<void> {
        await apiClient.post(`${this.basePath}/templates/${slug}/rate`, request);
    }

    async addComment(slug: string, request: AddCommentRequest): Promise<CommentDto> {
        const response = await apiClient.post(`${this.basePath}/templates/${slug}/comments`, request);
        return response.data;
    }

    async getTemplateStats(slug: string): Promise<TemplateStatsDto> {
        const response = await apiClient.get(`${this.basePath}/templates/${slug}/stats`);
        return response.data;
    }

    async getFeaturedTemplates(): Promise<TemplateDto[]> {
        const response = await apiClient.get(`${this.basePath}/templates/featured`);
        return response.data;
    }

    async getTrendingTemplates(period: string = 'week'): Promise<TemplateDto[]> {
        const response = await apiClient.get(`${this.basePath}/templates/trending`, {
            params: { period },
        });
        return response.data;
    }

    async getCategories(): Promise<CategoryDto[]> {
        const response = await apiClient.get(`${this.basePath}/categories`);
        return response.data;
    }

    async getPopularTags(limit: number = 50): Promise<TagDto[]> {
        const response = await apiClient.get(`${this.basePath}/tags`, {
            params: { limit },
        });
        return response.data;
    }

    // Organization endpoints
    async createOrganization(request: any): Promise<OrganizationDto> {
        const response = await apiClient.post(`${this.basePath}/organizations`, request);
        return response.data;
    }

    async getOrganization(slug: string): Promise<OrganizationDetailDto> {
        const response = await apiClient.get(`${this.basePath}/organizations/${slug}`);
        return response.data;
    }

    async getOrganizationTemplates(slug: string, page: number = 0, size: number = 20): Promise<Page<TemplateDto>> {
        const response = await apiClient.get(`${this.basePath}/organizations/${slug}/templates`, {
            params: { page, size },
        });
        return response.data;
    }

    // User endpoints
    async getUserTemplates(page: number = 0, size: number = 20): Promise<Page<TemplateDto>> {
        const response = await apiClient.get(`${this.basePath}/my/templates`, {
            params: { page, size },
        });
        return response.data;
    }

    async getUserInstallations(page: number = 0, size: number = 20): Promise<Page<any>> {
        const response = await apiClient.get(`${this.basePath}/my/installations`, {
            params: { page, size },
        });
        return response.data;
    }

    async uninstallTemplate(installationId: string): Promise<void> {
        await apiClient.delete(`${this.basePath}/my/installations/${installationId}`);
    }

    // Admin endpoints
    async certifyTemplate(slug: string): Promise<void> {
        await apiClient.post(`${this.basePath}/admin/templates/${slug}/certify`);
    }

    async featureTemplate(slug: string, duration: number): Promise<void> {
        await apiClient.post(`${this.basePath}/admin/templates/${slug}/feature`, { duration });
    }

    async verifyOrganization(slug: string): Promise<void> {
        await apiClient.post(`${this.basePath}/admin/organizations/${slug}/verify`);
    }
}

export const marketplaceApi = new MarketplaceService();

File: backend/src/main/java/com/integrixs/backend/marketplace/service/MarketplaceService.java
package com.integrixs.backend.marketplace.service;

import com.integrixs.backend.marketplace.dto.*;
// Entity and repository imports removed during SQL migration
import com.integrixs.backend.marketplace.exception.TemplateNotFoundException;
import com.integrixs.backend.marketplace.exception.UnauthorizedAccessException;
import com.integrixs.data.model.User;
import com.integrixs.backend.auth.service.AuthService;
import com.integrixs.backend.application.service.IntegrationFlowService;
import com.integrixs.data.model.IntegrationFlow;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.web.multipart.MultipartFile;

import java.io.IOException;
import java.time.LocalDateTime;
import java.util.*;
import java.util.ArrayList;
import java.util.stream.Collectors;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Service
public class MarketplaceService {

    private static final Logger log = LoggerFactory.getLogger(MarketplaceService.class);

    // Note: Marketplace functionality has been disabled after database migration.
    // All repository dependencies were removed as part of SQL migration.
    // This service currently returns empty results for all operations.

    private final IntegrationFlowService flowService;
    private final AuthService authService;
    private final FileStorageService fileStorageService;
    private final TemplateValidationService validationService;
    private final ObjectMapper objectMapper;

    // Constructor with reduced dependencies after SQL migration
    public MarketplaceService(IntegrationFlowService flowService,
                            AuthService authService,
                            FileStorageService fileStorageService,
                            TemplateValidationService validationService,
                            ObjectMapper objectMapper) {
        this.flowService = flowService;
        this.authService = authService;
        this.fileStorageService = fileStorageService;
        this.validationService = validationService;
        this.objectMapper = objectMapper;
    }

    /**
     * Search templates with filters
     */
    public Page<TemplateDto> searchTemplates(TemplateSearchRequest request, Pageable pageable) {
        // Marketplace functionality has been removed due to database entity deletion
        // Return empty results until marketplace is reimplemented with SQL
        log.info("Marketplace search requested but marketplace entities have been removed");
        return Page.empty(pageable);
    }

    /**
     * Get template details
     */
    public TemplateDetailDto getTemplateDetails(String slug) {
        log.info("Marketplace getTemplateDetails requested but marketplace functionality is disabled");
        throw new TemplateNotFoundException("Marketplace functionality is currently disabled");
    }

    /**
     * Create a new template
     */
    public TemplateDto createTemplate(CreateTemplateRequest request) {
        log.info("Marketplace createTemplate requested but marketplace functionality is disabled");
        return TemplateDto.builder()
            .id(UUID.randomUUID())
            .name(request.getName())
            .description("Marketplace functionality is currently disabled")
            .build();
    }

    /**
     * Update template
     */
    public TemplateDto updateTemplate(String slug, UpdateTemplateRequest request) {
        log.info("Marketplace updateTemplate requested but marketplace functionality is disabled");
        return TemplateDto.builder()
            .id(UUID.randomUUID())
            .slug(slug)
            .name(request.getName())
            .description("Marketplace functionality is currently disabled")
            .build();
    }

    /**
     * Upload template icon
     */
    public void uploadIcon(String slug, MultipartFile file) throws IOException {
        log.info("Marketplace uploadIcon requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

    /**
     * Add template screenshot
     */
    public void addScreenshot(String slug, MultipartFile file) throws IOException {
        log.info("Marketplace addScreenshot requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

    /**
     * Publish new version
     */
    public TemplateVersionDto publishVersion(String slug, PublishVersionRequest request) {
        log.info("Marketplace publishVersion requested but marketplace functionality is disabled");
        return TemplateVersionDto.builder()
            .version("0.0.0")
            .releaseNotes("Marketplace functionality is currently disabled")
            .build();
    }

    /**
     * Install template
     */
    public InstallationResultDto installTemplate(String slug, InstallTemplateRequest request) {
        log.info("Marketplace installTemplate requested but marketplace functionality is disabled");
        return InstallationResultDto.builder()
            .success(false)
            .message("Marketplace functionality is currently disabled")
            .build();
    }

    /**
     * Rate template
     */
    public void rateTemplate(String slug, RateTemplateRequest request) {
        log.info("Marketplace rateTemplate requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

    /**
     * Add comment to template
     */
    public CommentDto addComment(String slug, AddCommentRequest request) {
        log.info("Marketplace addComment requested but marketplace functionality is disabled");
        return CommentDto.builder()
            .id(UUID.randomUUID())
            .content("Marketplace functionality is currently disabled")
            .build();
    }

    /**
     * Get featured templates
     */
    public List<TemplateDto> getFeaturedTemplates() {
        log.info("Marketplace getFeaturedTemplates requested but marketplace functionality is disabled");
        return new ArrayList<>();
    }

    /**
     * Get trending templates
     */
    public List<TemplateDto> getTrendingTemplates(String period) {
        log.info("Marketplace getTrendingTemplates requested but marketplace functionality is disabled");
        return new ArrayList<>();
    }

    /**
     * Get template statistics
     */
    public TemplateStatsDto getTemplateStats(String slug) {
        log.info("Marketplace getTemplateStats requested but marketplace functionality is disabled");
        return TemplateStatsDto.builder()
            .downloadCount(0L)
            .installCount(0L)
            .averageRating(0.0)
            .ratingCount(0L)
            .versionCount(0L)
            .commentCount(0L)
            .lastUpdated(LocalDateTime.now())
            .build();
    }

    // Helper methods - all removed due to entity dependencies

    // DTO conversion methods - all removed due to entity dependencies

    /**
     * Get all template categories with counts
     */
    public List<CategoryDto> getCategories() {
        // Implementation to get categories with template counts
        List<CategoryDto> categories = new ArrayList<>();

        // This would typically query the database for category counts
        // For now, returning placeholder implementation
        return categories;
    }

    /**
     * Get popular tags used in templates
     */
    public List<TagDto> getPopularTags(int limit) {
        // Implementation to get popular tags
        List<TagDto> tags = new ArrayList<>();

        // This would typically query the database for tag usage counts
        // For now, returning placeholder implementation
        return tags;
    }

    /**
     * Get user's templates
     */
    public Page<TemplateDto> getUserTemplates(Pageable pageable) {
        log.info("Marketplace getUserTemplates requested but marketplace functionality is disabled");
        return Page.empty(pageable);
    }

    /**
     * Get user's installations
     */
    public Page<InstallationDto> getUserInstallations(Pageable pageable) {
        log.info("Marketplace getUserInstallations requested but marketplace functionality is disabled");
        return Page.empty(pageable);
    }

    /**
     * Uninstall template
     */
    public void uninstallTemplate(UUID installationId) {
        log.info("Marketplace uninstallTemplate requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

    /**
     * Certify template (admin only)
     */
    public void certifyTemplate(String slug) {
        log.info("Marketplace certifyTemplate requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

    /**
     * Feature template (admin only)
     */
    public void featureTemplate(String slug, int duration) {
        log.info("Marketplace featureTemplate requested but marketplace functionality is disabled");
        // Marketplace functionality is disabled - no action taken
    }

}
